lib/spack/spack/build_systems/cmake.py

```diff
@@ -38,19 +38,36 @@ class CMakePackage(PackageBase):
 
     This class provides three phases that can be overridden:
 
-    * cmake
-    * build
-    * install
+        1. :py:meth:`~.CMakePackage.cmake`
+        2. :py:meth:`~.CMakePackage.build`
+        3. :py:meth:`~.CMakePackage.install`
 
     They all have sensible defaults and for many packages the only thing
-    necessary will be to override ``cmake_args``
+    necessary will be to override :py:meth:`~.CMakePackage.cmake_args`.
+    For a finer tuning you may override:
+
+        +-----------------------------------------------+--------------------+
+        | **Method**                                    | **Purpose**        |
+        +===============================================+====================+
+        | :py:meth:`~.CMakePackage.build_type`          | Specify the value  |
+        |                                               | for the            |
+        |                                               | CMAKE_BUILD_TYPE   |
+        |                                               | variable           |
+        +-----------------------------------------------+--------------------+
+        | :py:meth:`~.CMakePackage.root_cmakelists_dir` | Directory where to |
+        |                                               | find the root      |
+        |                                               | CMakeLists.txt     |
+        +-----------------------------------------------+--------------------+
+        | :py:meth:`~.CMakePackage.build_directory`     | Directory where to |
+        |                                               | build the package  |
+        +-----------------------------------------------+--------------------+
+
 
-    Additionally, you may specify make targets for build and install
-    phases by overriding ``build_targets`` and ``install_targets``
     """
+    #: Phases of a CMake package
     phases = ['cmake', 'build', 'install']
-    # To be used in UI queries that require to know which
-    # build-system class we are using
+    #: This attribute is used in UI queries that require to know which
+    #: build-system class we are using
```

Comment:
"This attribute is used by UI queries that need to know the build system base class."